[veSync] 131 and Vital Purifiers base support (#15296)
openhab/openhab-addons: bundles/org.openhab.binding.vesync/README.md

This binding was developed from the great work in the listed projects.

> The only Air Filter unit it has been tested against are the Core400S unit and Vital 100S, **I'm looking for others to confirm** my queries regarding **the Core200S and Core300S** units.

Reviewer: ```suggestion
The binding has been tested against the following Air Filter unit's: Core400S and Vital 100S, **I'm looking for others to confirm** my queries regarding **the Core200S and Core300S** units.
```

| filterLifePercentage | Number:Dimensionless | The remaining filter life as a percentage                  | 600S, 400S, 300S  |                       |
| airQuality           | Number:Dimensionless | The air quality as represented by the Core200S / Core300S  | 600S, 400S, 300S  |                       |
| airQualityPM25       | Number:Density       | The air quality as represented by the Core400S             | 600S, 400S, 300S  |                       |
| errorCode            | Number:Dimensionless | The error code reported by the device                      | 600S, 400S, 300S  |                       |
| timerExpiry          | DateTime             | The expected expiry time of the current timer              | 600S, 400S        |                       |
| schedulesCount       | Number:Dimensionless | The number schedules configured                            | 600S, 400S        |                       |
| configDisplayForever | Switch               | Config: Whether the display will disable when not active   | 600S, 400S, 300S  |                       |

> | configAutoMode       | String               | Config: The mode of operation when auto is active          | 600S, 400S, 300S  |                       |

Author: Yeah I think this was removed intentionally - so long ago I am struggling to remember now the reverse engineering of the python lib, this is based from. I will have another look at that to check if warm node is supported and understand this, while this soak tests with the adjustments. Will come back to this one, once I've had a couple of weeks (sorry works manic) to figure it all out.

> | configAutoMode       | String               | Config: The mode of operation when auto is active          | 600S, 400S, 300S  |                       |

Reviewer: Oh just take your time. As long as you manage to be sure and we can make the use of this consistent.

> | configAutoMode       | String               | Config: The mode of operation when auto is active          | 600S, 400S, 300S  |                       |

Author: I take back the last statement I think that was a mistake after cross checking it. I have just pushed a load of minor adjustments, to clean up little bits, and added that back in. Will continue checking tomorrow, but that likely was a mistake on my part. I'll have another look tomorrow in case I can see a reason why it was removed, but at the moment I cant spot it.

| **humidifierMode**         | String               | The current mode of operation                                 | 200S, Dual200S, 300S, 600S, OasisMist, OasisMist1000 | [auto, sleep]       |      |
| **nightLightMode**         | String               | The night light mode                                          | 200S, Dual200S, 300S                                 | [on, dim, off]      |      |
| **humiditySetpoint**       | Number:Dimensionless | Humidity % set point to reach                                 | 200S, Dual200S, 300S, 600S, OasisMist, OasisMist1000 | [30...80]           |      |
| warmEnabled                | Switch               | Indicator for warm mist mode                                  | 600S, OasisMist                                      |                     |      |
| **warmLevel**              | Number:Dimensionless | The current warm mist level set                               | 600S, OasisMist                                      | [0..3]              | one  |

> | errorCode                  | Number:Dimensionless | The error code reported by the device                         | OasisMist1000                                        |                     | one  |

Reviewer: I would expect a ordinairy DecimalType, but if it needs to be QuantityType, please also adjust the item exmaple file (it lacks the Dimensionless part)

Author: I presume to keep everything aligned leave these as are now?

Reviewer: I think these channels should be fixed in a follow up PR.
